File: gin-vue-admin/web/src/api/starNote/userAccount.js
import service from '@/utils/request'
// @Tags UserAccount
// @Summary 创建用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.UserAccount true "创建用户账号"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /ua/createUserAccount [post]
export const createUserAccount = (data) => {
  return service({
    url: '/ua/createUserAccount',
    method: 'post',
    data
  })
}

// @Tags UserAccount
// @Summary 删除用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.UserAccount true "删除用户账号"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /ua/deleteUserAccount [delete]
export const deleteUserAccount = (params) => {
  return service({
    url: '/ua/deleteUserAccount',
    method: 'delete',
    params
  })
}

// @Tags UserAccount
// @Summary 批量删除用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除用户账号"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /ua/deleteUserAccount [delete]
export const deleteUserAccountByIds = (params) => {
  return service({
    url: '/ua/deleteUserAccountByIds',
    method: 'delete',
    params
  })
}

// @Tags UserAccount
// @Summary 更新用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.UserAccount true "更新用户账号"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /ua/updateUserAccount [put]
export const updateUserAccount = (data) => {
  return service({
    url: '/ua/updateUserAccount',
    method: 'put',
    data
  })
}

// @Tags UserAccount
// @Summary 用id查询用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query model.UserAccount true "用id查询用户账号"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /ua/findUserAccount [get]
export const findUserAccount = (params) => {
  return service({
    url: '/ua/findUserAccount',
    method: 'get',
    params
  })
}

// @Tags UserAccount
// @Summary 分页获取用户账号列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取用户账号列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /ua/getUserAccountList [get]
export const getUserAccountList = (params) => {
  return service({
    url: '/ua/getUserAccountList',
    method: 'get',
    params
  })
}

// @Tags UserAccount
// @Summary 不需要鉴权的用户账号接口
// @Accept application/json
// @Produce application/json
// @Param data query starNoteReq.UserAccountSearch true "分页获取用户账号列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /ua/getUserAccountPublic [get]
export const getUserAccountPublic = () => {
  return service({
    url: '/ua/getUserAccountPublic',
    method: 'get',
  })
}

export const getAdminTags = () => {
  return service({
    url: '/admin/tags',
    method: 'get',
  })
}

export const getAdminUserTags = (userId) => {
  return service({
    url: `/admin/users/${userId}/tags`,
    method: 'get',
  })
}

export const updateAdminUserTags = (userId, data) => {
  return service({
    url: `/admin/users/${userId}/tags`,
    method: 'post',
    data,
  })
}

File: gin-vue-admin/rm_file/349632000/server/api/v1/starNote/user_account.go
package starNote

import (
	
	"github.com/flipped-aurora/gin-vue-admin/server/global"
    "github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
    "github.com/flipped-aurora/gin-vue-admin/server/model/starNote"
    starNoteReq "github.com/flipped-aurora/gin-vue-admin/server/model/starNote/request"
    "github.com/gin-gonic/gin"
    "go.uber.org/zap"
)

type UserAccountApi struct {}



// CreateUserAccount 创建用户账号
// @Tags UserAccount
// @Summary 创建用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body starNote.UserAccount true "创建用户账号"
// @Success 200 {object} response.Response{msg=string} "创建成功"
// @Router /ua/createUserAccount [post]
func (uaApi *UserAccountApi) CreateUserAccount(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	var ua starNote.UserAccount
	err := c.ShouldBindJSON(&ua)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = uaService.CreateUserAccount(ctx,&ua)
	if err != nil {
        global.GVA_LOG.Error("创建失败!", zap.Error(err))
		response.FailWithMessage("创建失败:" + err.Error(), c)
		return
	}
    response.OkWithMessage("创建成功", c)
}

// DeleteUserAccount 删除用户账号
// @Tags UserAccount
// @Summary 删除用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body starNote.UserAccount true "删除用户账号"
// @Success 200 {object} response.Response{msg=string} "删除成功"
// @Router /ua/deleteUserAccount [delete]
func (uaApi *UserAccountApi) DeleteUserAccount(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	ID := c.Query("ID")
	err := uaService.DeleteUserAccount(ctx,ID)
	if err != nil {
        global.GVA_LOG.Error("删除失败!", zap.Error(err))
		response.FailWithMessage("删除失败:" + err.Error(), c)
		return
	}
	response.OkWithMessage("删除成功", c)
}

// DeleteUserAccountByIds 批量删除用户账号
// @Tags UserAccount
// @Summary 批量删除用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Success 200 {object} response.Response{msg=string} "批量删除成功"
// @Router /ua/deleteUserAccountByIds [delete]
func (uaApi *UserAccountApi) DeleteUserAccountByIds(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	IDs := c.QueryArray("IDs[]")
	err := uaService.DeleteUserAccountByIds(ctx,IDs)
	if err != nil {
        global.GVA_LOG.Error("批量删除失败!", zap.Error(err))
		response.FailWithMessage("批量删除失败:" + err.Error(), c)
		return
	}
	response.OkWithMessage("批量删除成功", c)
}

// UpdateUserAccount 更新用户账号
// @Tags UserAccount
// @Summary 更新用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body starNote.UserAccount true "更新用户账号"
// @Success 200 {object} response.Response{msg=string} "更新成功"
// @Router /ua/updateUserAccount [put]
func (uaApi *UserAccountApi) UpdateUserAccount(c *gin.Context) {
    // 从ctx获取标准context进行业务行为
    ctx := c.Request.Context()

	var ua starNote.UserAccount
	err := c.ShouldBindJSON(&ua)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = uaService.UpdateUserAccount(ctx,ua)
	if err != nil {
        global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败:" + err.Error(), c)
		return
	}
	response.OkWithMessage("更新成功", c)
}

// FindUserAccount 用id查询用户账号
// @Tags UserAccount
// @Summary 用id查询用户账号
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param ID query uint true "用id查询用户账号"
// @Success 200 {object} response.Response{data=starNote.UserAccount,msg=string} "查询成功"
// @Router /ua/findUserAccount [get]
func (uaApi *UserAccountApi) FindUserAccount(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	ID := c.Query("ID")
	reua, err := uaService.GetUserAccount(ctx,ID)
	if err != nil {
        global.GVA_LOG.Error("查询失败!", zap.Error(err))
		response.FailWithMessage("查询失败:" + err.Error(), c)
		return
	}
	response.OkWithData(reua, c)
}
// GetUserAccountList 分页获取用户账号列表
// @Tags UserAccount
// @Summary 分页获取用户账号列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query starNoteReq.UserAccountSearch true "分页获取用户账号列表"
// @Success 200 {object} response.Response{data=response.PageResult,msg=string} "获取成功"
// @Router /ua/getUserAccountList [get]
func (uaApi *UserAccountApi) GetUserAccountList(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	var pageInfo starNoteReq.UserAccountSearch
	err := c.ShouldBindQuery(&pageInfo)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	list, total, err := uaService.GetUserAccountInfoList(ctx,pageInfo)
	if err != nil {
	    global.GVA_LOG.Error("获取失败!", zap.Error(err))
        response.FailWithMessage("获取失败:" + err.Error(), c)
        return
    }
    response.OkWithDetailed(response.PageResult{
        List:     list,
        Total:    total,
        Page:     pageInfo.Page,
        PageSize: pageInfo.PageSize,
    }, "获取成功", c)
}

// GetUserAccountPublic 不需要鉴权的用户账号接口
// @Tags UserAccount
// @Summary 不需要鉴权的用户账号接口
// @Accept application/json
// @Produce application/json
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /ua/getUserAccountPublic [get]
func (uaApi *UserAccountApi) GetUserAccountPublic(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

    // 此接口不需要鉴权
    // 示例为返回了一个固定的消息接口，一般本接口用于C端服务，需要自己实现业务逻辑
    uaService.GetUserAccountPublic(ctx)
    response.OkWithDetailed(gin.H{
       "info": "不需要鉴权的用户账号接口信息",
    }, "获取成功", c)
}
